cyksuid/_ksuid.pyi

```diff
@@ -0,0 +1,85 @@
+import functools
+from datetime import datetime
+from typing import Any, Callable, Optional, Type, TypeVar, overload
+
+_bytestr = bytes
+
+BYTE_LENGTH: int
+STRING_ENCODED_LENGTH: int
+EMPTY_BYTES: bytes
+MAX_ENCODED: bytes
+
+SelfT = TypeVar("SelfT", bound="Ksuid")
+
+@functools.total_ordering
+class Ksuid:
+    """KSUIDs are 20 bytes contains 4 byte timestamp with custom epoch and 16 bytes random data."""
+
+    BASE62_LENGTH: int
+    PAYLOAD_LENGTH_IN_BYTES: int
+    TIMESTAMP_LENGTH_IN_BYTES: int
+
+    @overload
+    def __init__(self) -> None:
+        """Create a new KSUID with current timestamp and generated random payload."""
+    @overload
+    def __init__(self, raw: bytes) -> None:
+        """Creates KSUID from raw bytes."""
+    @overload
+    def __init__(self, timestamp: int | float, payload: bytes) -> None:
+        """Creates KSUID from specified timestamp in milliseconds and payload."""
+    @overload
+    def __init__(self, **kwargs: Any) -> None: ...
+    @classmethod
+    def from_payload(cls: Type[SelfT], payload: bytes) -> SelfT: ...
```

Comment:
Would it make sense to also have a `from_timestamp` method? Seems like the generic `ksuid` function supports that.

Reply:
To add some context, `from_timestamp` would be useful if we want to retroactively create `Ksuid`s for records where we have a timestamp. It's easily achievable using the `ksuid(time_func=...)`, but would be nice to have as a static method on the classes too.

Reply:
> Would it make sense to also have a `from_timestamp` method? Seems like the generic `ksuid` function supports that.

Make sense, I'll implement that later